shell/platform/android/io/flutter/view/ResourceExtractor.java

```diff
@@ -339,4 +346,114 @@ private String checkTimestamp(File dataDir) {
 
         return null;
     }
+
+    private static void copy(InputStream in, OutputStream out) throws IOException {
+        byte[] buf = new byte[16 * 1024];
+        for (int i; (i = in.read(buf)) >= 0; ) {
+            out.write(buf, 0, i);
+        }
+    }
+
+    private static void read(InputStream in, byte[] buf, int off, int len) throws IOException {
+        for (int i, n = 0; n < len; n += i) {
+            if ((i = in.read(buf, off+n, len-n)) < 0) {
```

Comment:
nit: add spaced around the `+` and `-` operators

Reply:
Done